api/ccapi/include/layer.h

```diff
@@ -363,6 +364,23 @@ CrossEntropy(const std::vector<std::string> &properties = {});
 
 } // namespace loss
 
+/**
+ * @brief General Layer Factory function to register Layer
+ *
+ * @param props property representation
+ * @return std::unique_ptr<ml::train::Layer> created object
+ */
+template <typename T,
```

Comment:
Let's add a comment in `layer_internal` and `optimizer_internal` to ensure to make a unit test with `createLayer/createOptimizer`, otherwise, their templated function won't be made.

Reply:
Isn't the reference made because `createLayer/createOptimizer` is used in `app_context.cpp`?

Please correct me if I have made a misunderstanding

Reply:
Then a comment that all new layers to be used must be included in `app_context.cpp`.

Reply:
Done